src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1688,6 +1691,73 @@ Expression* TranslateToFuzzReader::makeTry(Type type) {
   return builder.makeTry(body, catchTags, catchBodies);
 }
 
+Expression* TranslateToFuzzReader::makeTryTable(Type type) {
+  if (funcContext->breakableStack.empty()) {
+    // Nothing to break to.
+    // TODO: Perhaps generate a block wrapping us?
+    return makeTrivial(type);
+  }
+  if (wasm.tags.empty()) {
+    addTag();
+  }
+
+  // Add catches of specific tags, and possibly a catch_all at the end. We use
+  // the last iteration of the loop for that.
+  std::vector<Name> catchTags;
+  std::vector<Name> catchDests;
+  std::vector<bool> catchRefs;
+  auto numTags = upTo(MAX_TRY_CATCHES);
+  for (Index i = 0; i <= numTags; i++) {
+    Name tagName;
+    Type tagType;
+    if (i < numTags) {
+      // Look for a specific tag.
+      auto& tag = pick(wasm.tags);
+      tagName = tag->name;
+      tagType = tag->sig.params;
+    } else {
+      // Look for a catch_all at the end, some of the time (but all of the time
+      // if we have nothing else).
+      if (!catchTags.empty() && oneIn(2)) {
+        break;
+      }
+      tagType = Type::none;
+    }
+
+    // We need to find a proper target to break to, which means a target that
+    // has the type of the tag, or the tag + an exnref at the end.
+    std::vector<Type> vec;
+    for (auto t : tagType) {
+      vec.push_back(t);
+    }
+    vec.push_back(Type(HeapType::exn, Nullable)); // TODO: NonNullable?
+    auto tagTypeWithExn = Type(vec);
+    int tries = TRIES;
+    while (tries-- > 0) {
+      auto* target = pick(funcContext->breakableStack);
+      auto name = getTargetName(target);
+      auto valueType = getTargetType(target);
+      if (valueType == tagType || valueType == tagTypeWithExn) {
+        catchTags.push_back(tagName);
+        catchDests.push_back(name);
```

Comment:
Maybe `destName` or `dest` or something to be distinguished from `tagName`?

Reply:
Renamed.